qiita_db/test/test_data.py

```diff
@@ -478,6 +478,38 @@ def test_create_data_type_only(self):
         # preprocessed_data_id, filepath_id
         self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
 
+    def test_delete(self):
+        """Correctly deletes a preprocessed data"""
+        # testing regular delete
+        ppd = PreprocessedData.create(
+            self.study, self.params_table,
+            self.params_id, self.filepaths, prep_template=self.prep_template,
+            ebi_submission_accession=self.ebi_submission_accession,
+            ebi_study_accession=self.ebi_study_accession)
+        PreprocessedData.delete(ppd.id)
```

Comment:
That makes a bit more sense. Nevertheless, the first thing we do when trying to delete any object is to check that the object exists by instantiate it, which we do cause we need to check different status to allow delete. In fact, if you see the test is that it raises a QiitaDBUnknownIDError, which is raised by the instantiation.